### PR TITLE
Fixed #30585 -- Added {% translate %} and {% blocktranslate %} template tags.

### PR DESCRIPTION
**Pull request by mwhansen**
https://code.djangoproject.com/ticket/30585

https://groups.google.com/forum/#!msg/django-developers/tANKEpROLeQ/mbXwb482DwAJ

### CODE REVIEW

File: django/templatetags/i18n.py [hunk 1]
```diff
@@ -98,7 +98,8 @@ def render(self, context):
 class BlockTranslateNode(Node):
 
     def __init__(self, extra_context, singular, plural=None, countervar=None,
-                 counter=None, message_context=None, trimmed=False, asvar=None):
+                 counter=None, message_context=None, trimmed=False, asvar=None,
+                 tag_name='blocktrans'):
```
Comment (carltongibson) [inline django/templatetags/i18n.py]: Why are we preferring the old alias here?

**Comment by carltongibson** [comment]
So I pushed d3fa0dedbaa21606144fc3d85a983ca4094b747e which reverts the change to `blocktrans` for one test example, thereby fixing one failure. Obviously that's not for merging. Rather it just shows there's more to be done.

File: django/templatetags/i18n.py (hunk 1 above)
Comment (mwhansen) [reply django/templatetags/i18n.py]: I think just a case I missed.  -- I'll update that to use the new alias.

**Comment by mwhansen** [comment]
Thanks for the help!  I'll dig into the errors today.

**Comment by mwhansen** [comment]
@carltongibson I push a commit which should fix the remaining issues.

**Comment by carltongibson** [comment]
@mwhansen Super. Thanks. I shall look again tomorrow.

**Comment by felixxm** [comment]
@mwhansen Thanks, great job :rocket: I pushed some changes:

- wrapped docs at 79 chars,
- reverted unrelated wrapping in docs,
- used `inspect.signature()` instead of deprecated `inspect.getargspec()`,
- remove unnecessary: `# coding: utf-8`,

- moved `tests/template_tests/syntax_tests/i18n/test_trans.py` to `tests/template_tests/syntax_tests/i18n/test_translate.py`.

**Comment by mwhansen** [comment]
Thanks @felixxm !

**Comment by aq1** [comment]
White people are crazy

**Pull request merged**